### PR TITLE
.Net & Python: Fix Broken Links

### PR DESCRIPTION
**Pull request by justindhillon**
### Description

I used [link-inspector](https://github.com/justindhillon/link-inspector) to find and fix broken links in this project. This is an updated PR of  #4910, which fixes the merge conflict. Here are the links I have fixed:

https://docs.microsoft.com/rest/api/keyvault/getsecrets/getsecrets --> https://learn.microsoft.com/en-us/rest/api/keyvault/secrets/get-secret/get-secret

https://docs.microsoft.com/rest/api/keyvault/getkeys/getkeys --> https://learn.microsoft.com/en-us/rest/api/keyvault/keys/get-keys/get-keys

https://docs.microsoft.com/rest/api/keyvault/encrypt/encrypt --> https://learn.microsoft.com/en-us/rest/api/keyvault/keys/encrypt/encrypt

https://docs.microsoft.com/rest/api/keyvault/decrypt/decrypt --> https://learn.microsoft.com/en-us/rest/api/keyvault/keys/decrypt/decrypt

https://docs.microsoft.com/rest/api/keyvault/createkey/createkey --> https://learn.microsoft.com/en-us/rest/api/keyvault/keys/create-key/create-key

https://docs.microsoft.com/rest/api/keyvault/getsecretversions/getsecretversions --> https://learn.microsoft.com/en-us/rest/api/keyvault/secrets/get-secret-versions/get-secret-versions

https://github.com/microsoft/semantic-kernel/blob/main/python/semantic_kernel/planning/plan.py --> https://github.com/microsoft/semantic-kernel/blob/main/python/semantic_kernel/planners/plan.py

https://github.com/microsoft/semantic-kernel/blob/main/python/semantic_kernel/planning/sequential_planner/Plugins/SequentialPlanning/skprompt.txt --> https://github.com/microsoft/semantic-kernel/blob/main/samples/plugins/QAPlugin/Form/skprompt.txt

### Support my work

These links were found with [link-inspector](https://github.com/justindhillon/link-inspector). If you find this PR useful, give the repo a ⭐

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Review by markwallace-microsoft — COMMENTED** [review]
One of the changes is incorrect

File: python/notebooks/05-using-the-planner.ipynb [hunk 1]
```diff
@@ -367,7 +367,7 @@
    "id": "a1c66d83",
    "metadata": {},
    "source": [
-    "The sequential planner is an XML-based step-by-step planner. You can see the prompt used for it here (https://github.com/microsoft/semantic-kernel/blob/main/python/semantic_kernel/planning/sequential_planner/Plugins/SequentialPlanning/skprompt.txt)\n"
+    "The sequential planner is an XML-based step-by-step planner. You can see the prompt used for it here (https://github.com/microsoft/semantic-kernel/blob/main/samples/plugins/QAPlugin/Form/skprompt.txt)\n"
```
[the remark this answers is not in the document]
Comment (justindhillon) [reply python/notebooks/05-using-the-planner.ipynb]: Thanks for the feedback! The link has been replaced with the correct link.

**Comment by justindhillon** [comment]
Based on @markwallace-microsoft  feedback, I have fixed this link: 

https://github.com/microsoft/semantic-kernel/blob/main/samples/plugins/QAPlugin/Form/skprompt.txt --> https://github.com/microsoft/semantic-kernel/blob/main/python/semantic_kernel/planners/sequential_planner/Plugins/SequentialPlanning/skprompt.txt

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/eac012984b4ed507d10c04f47a24b2d71bce30f0/README.md">Py3.10 Test Coverage</a><details><summary>Python 3.10 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>5187</b></td><td><b>982</b></td><td><b>81%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python 3.10 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1059 | 1 :zzz: | 0 :x: | 0 :fire: | 28.566s :stopwatch: |

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/eac012984b4ed507d10c04f47a24b2d71bce30f0/README.md">Py3.8 Test Coverage</a><details><summary>Python 3.8 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>5044</b></td><td><b>990</b></td><td><b>80%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python 3.8 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1059 | 11 :zzz: | 0 :x: | 0 :fire: | 19.915s :stopwatch: |

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/eac012984b4ed507d10c04f47a24b2d71bce30f0/README.md">Py3.11 Test Coverage</a><details><summary>Python 3.11 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>5187</b></td><td><b>982</b></td><td><b>81%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python 3.11 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1059 | 1 :zzz: | 0 :x: | 0 :fire: | 23.620s :stopwatch: |

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/eac012984b4ed507d10c04f47a24b2d71bce30f0/README.md">Py3.9 Test Coverage</a><details><summary>Python 3.9 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>5181</b></td><td><b>984</b></td><td><b>81%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python 3.9 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1059 | 2 :zzz: | 0 :x: | 0 :fire: | 25.240s :stopwatch: |

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/eac012984b4ed507d10c04f47a24b2d71bce30f0/README.md">Py3.12 Test Coverage</a><details><summary>Python 3.12 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>5187</b></td><td><b>982</b></td><td><b>81%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python 3.12 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1059 | 1 :zzz: | 0 :x: | 0 :fire: | 32.281s :stopwatch: |

**Comment by dmytrostruk** [comment]
@moonbox3 @eavanvalkenburg Would it be possible to move code coverage reports for Python code from PR comments to CI pipeline? Posting code coverage in PR comments has some disadvantages:
1. Code coverage comments are mixed with user comments in PR which makes it harder to review.
2. With every code coverage comment for each Python version I receive separate email notification.

For .NET code, we output code coverage in CI pipeline. It's useful when code coverage threshold is not met and you can see which assembly requires more tests. It's also possible to check code coverage for specific .NET SDK version:
image

Maybe we can do something similar for Python? Let me know what you think, thanks!

**Review by eavanvalkenburg — APPROVED** [review]
python side looks good

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**